cmd/register/register_util.go

```diff
@@ -414,27 +441,97 @@ func getJSONSpec(message proto.Message) string {
 	opt := &github.ListOptions{Page: 1, PerPage: 1}
-	releases, _, err := c.Repositories.ListReleases(context.Background(), githubOrg, githubRepository, opt)
+	releases, _, err := c.Repositories.ListReleases(context.Background(), org, repository, opt)
 	if err != nil {
 		return nil, "", err
 	}
-	response, err := http.Get(fmt.Sprintf(flyteManifest, *releases[0].TagName))
+	if len(releases) > 0 {
+		response, err := http.Get(fmt.Sprintf(flyteManifest, *releases[0].TagName))
+		if err != nil {
+			return nil, "", err
+		}
+		defer response.Body.Close()
+
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, "", err
+		}
+
+		err = json.Unmarshal(data, &FlyteSnacksRelease)
+		if err != nil {
+			return nil, "", err
+		}
+		return FlyteSnacksRelease, *releases[0].TagName, nil
+	}
+	return nil, "", fmt.Errorf("Repository doesn't have any release")
+}
+
+func getAdditionalDistributionLoc(remoteLocation, identifier string) storage.DataReference {
+	return storage.DataReference(fmt.Sprintf("%v/%v.tar.gz", remoteLocation, identifier))
+}
+
+func uploadFastRegisterArtifact(ctx context.Context, file, additionalDistributionDir, version string) error {
+	s, err := getStorageClient(ctx)
 	if err != nil {
-		return nil, "", err
+		return err
 	}
-	defer response.Body.Close()
 
-	data, err := ioutil.ReadAll(response.Body)
+	var dataRefReaderCloser io.ReadCloser
+	fullRemotePath := getAdditionalDistributionLoc(additionalDistributionDir, version)
+	raw, err := json.Marshal(file)
 	if err != nil {
-		return nil, "", err
+		return err
+	}
+	dataRefReaderCloser, err = os.Open(file)
+	if err != nil {
+		return err
+	}
+	dataRefReaderCloser, err = gzip.NewReader(dataRefReaderCloser)
+	if err != nil {
+		return err
+	}
+	if err := s.WriteRaw(ctx, fullRemotePath, int64(len(raw)), storage.Options{}, dataRefReaderCloser); err != nil {
+		return err
 	}
+	return nil
+}
 
-	err = json.Unmarshal(data, &FlyteSnacksRelease)
+func getStorageClient(ctx context.Context) (storage.ComposedProtobufStore, error) {
+	if Client != nil {
+		return Client, nil
+	}
+	testScope := promutils.NewTestScope()
+	// Set Keys
+	labeled.SetMetricKeys(contextutils.AppNameKey, contextutils.ProjectKey, contextutils.DomainKey)
+	s, err := storage.NewDataStore(storage.GetConfig(), testScope.NewSubScope("flytectl"))
 	if err != nil {
-		return nil, "", err
+		logger.Errorf(ctx, "error while creating storage client %v", err)
+		return nil, err
+	}
+	return s.ComposedProtobufStore, nil
+}
+
+func validateRegisterFiles(dataRefs []string) (string, []string, []string, error) {
+	var validProto, InvalidFiles []string
+	var sourceCode string
+	var isFast = false
+	for _, v := range dataRefs {
+		if len(rconfig.DefaultFilesConfig.AdditionalDistributionPath) > 0 && strings.HasSuffix(v, sourceCodeExtension) {
+			isFast = true
+			sourceCode = v
+		} else if len(rconfig.DefaultFilesConfig.AdditionalDistributionPath) == 0 && strings.HasSuffix(v, sourceCodeExtension) {
```

Comment:
Instead of this check why can we not use the fast register regex.
And also whether a tar is fast registered tar can independently be determined just by its name and we need not use flag.